libcontainer/rootfs_linux.go

```diff
@@ -88,11 +86,19 @@ func setupRootfs(config *configs.Config, console *linuxConsole, pipe io.ReadWrit
 	if err != nil {
 		return newSystemErrorWithCause(err, "jailing process inside rootfs")
 	}
+
 	if setupDev {
 		if err := reOpenDevNull(); err != nil {
 			return newSystemErrorWithCause(err, "reopening /dev/null inside container")
 		}
 	}
+
+	return nil
+}
+
+// finalizeRootfs actually switches the root of the process and sets anything
+// to ro if necessary. You must call prepareRootfs first.
+func finalizeRootfs(config *configs.Config) (err error) {
```

Comment:
Yes. "Finalize" really means to "mount things as ro as requested". My original patchset had `PivotRoot` in `finalizeRootfs` but I ran into a bunch of permission issues (and the code had to mess around with paths quite a bit).